realsense2_camera/src/base_realsense_node.cpp

```diff
@@ -1707,16 +1713,17 @@ void BaseRealSenseNode::frame_callback(rs2::frame frame)
 
                 if (_enable_bright_region_removal)
                 {
-                    rs2::frameset fs = frame.as<rs2::frameset>();
-                    auto ir = fs.as<rs2::frameset>().get_infrared_frame(1);
-                    if (ir)
+                    static int count_total = 0, count_skipped = 0; 
+                    count_total++;
+                    if (infrared1_frame)
                     {
                         // Bright region removal
                         // TODO: to plugin?
-                        bright_removed = remove_bright_regions(depth_frame, ir);
+                        bright_removed = remove_bright_regions(depth_frame, infrared1_frame);
                     }
                     else{
-                        ROS_WARN("Skip bright region depth removal (No infra1 frame received)");
+                        count_skipped++;
+                        ROS_DEBUG_STREAM("Skip bright region depth removal (No infra1 frame received)." << " Skipped " << count_skipped << "/" << count_total << "frames.");
```

Comment:
This can possibly overflow maximum int depending on how long we run it. Suggest resetting the counters if count_total reaches std::numeric_limits<int>::max()

Reply:
@30fps, this will overflow in > 4.5 years of continuous operation. Still, will add the limit check.

Reply:
Hey with a static unsigned long, I think you can remove the check since the check is unnecessary as you mentioned. Sorry for the confusion.

Reply:
I think it would make sense to also print a percentage, because the ratio will become unreadable after a while if there are frames skipped sometimes

Reply:
Makes sense @ChristofDubs , I will make this change then merge the PR.